### PR TITLE
Failed Test Brief Log

### PR DESCRIPTION
**Pull request by kuznetsovmoci**
Need the ability to quickly look at the output for failing tests to determine the root cause.

This view lists all of the platforms containing a failing test.
Consolidates information found across platforms as seen in the Brief view under Tests

### CODE REVIEW

**Review by iguessthislldo — APPROVED** [review]

**Comment by jwillemsen** [comment]
It is not clear to me what this exactly adds, is it a page with just the test part of the full log in a separate file (so the part of the file starting at http://scoreboard.ociweb.com/dds_doc_Latest_Micro_bee_win7_vc12_i1d0o1s1m1/2021_02_16_03_58_Full.html#section_6 and further), or something else?

**Comment by kuznetsovmoci** [comment]
> It is not clear to me what this exactly adds, is it a page with just the test part of the full log in a separate file (so the part of the file starting at http://scoreboard.ociweb.com/dds_doc_Latest_Micro_bee_win7_vc12_i1d0o1s1m1/2021_02_16_03_58_Full.html#section_6 and further), or something else?

This new report combines individual "Brief" reports for all failed Tests, grouped by the server. It is intended to be a one-click summary of all failed tests. It is a time-saving alternative to clicking through all Test "Brief" links on the main page.

image

**Comment by jwillemsen** [comment]
How does this work with scoreboards that don't pull all logs locally, for example http://www.dre.vanderbilt.edu/scoreboard/integrated.html only has links and doesn't cache all logs of all builds, those are stored at the various locations, with a lot of builds (including huge valgrind logs) it costs a lot of bandwidth and diskspace, it uses the `-b` option of the scoreboard script.

**Comment by jwillemsen** [comment]
For the git commit you could use diffroot to construct a hyperlink to github, see the ACE/TAO scoreboard

**Comment by mitza-oci** [comment]
> 
> 
> How does this work with scoreboards that don't pull all logs locally, for example http://www.dre.vanderbilt.edu/scoreboard/integrated.html only has links and doesn't cache all logs of all builds, those are stored at the various locations, with a lot of builds (including huge valgrind logs) it costs a lot of bandwidth and diskspace, it uses the `-b` option of the scoreboard script.

Seems like the new feature could just be disabled when `-b` is used.

**Comment by kuznetsovmoci** [comment]
> > How does this work with scoreboards that don't pull all logs locally, for example http://www.dre.vanderbilt.edu/scoreboard/integrated.html only has links and doesn't cache all logs of all builds, those are stored at the various locations, with a lot of builds (including huge valgrind logs) it costs a lot of bandwidth and diskspace, it uses the `-b` option of the scoreboard script.
> 
> Seems like the new feature could just be disabled when `-b` is used.

Implemented

File: common/prettify.pm [hunk 1]
```diff
@@ -353,6 +353,190 @@ sub Normal ($)
...
+}
+
+sub Print_Sections ()
+{
+    my $self = shift;
+
+    if (defined $self->{LAST_SECTION} && defined $self->{LAST_SUBSECTION} && $self->{LAST_SECTION} eq 'Test') {
+        if (defined $self->{BUILDNAME}) {
+            print {$self->{FH}} "<hr><h2>$self->{BUILDNAME}</h2>\n";
+            my $rev = substr($self->{GIT_CHECKEDOUT_OPENDDS}, 0, 8);
```
Comment (jwillemsen) [inline common/prettify.pm:476]: This should be more flexible, some scoreboards don't have OpenDDS at all, at that moment only printing the OpenDDS git revision looks to be confusing. As quick fix maybe only print it when the revision is not unknown, best would be to print all git commit revisions that are known at that point.

File: common/prettify.pm [hunk 2]
```diff
@@ -353,6 +353,192 @@ sub Normal ($)
...
+            if ($rev ne "unknown") {
+                print {$self->{FH}} "Rev: <a href='https://github.com/objectcomputing/OpenDDS/commit/$rev' >$rev</a><hr>\n";
```
Comment (iguessthislldo) [inline common/prettify.pm:478]: We can't hardcore this to out repo. It's going to have to be a configuration option.

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: common/prettify.pm (hunk 2 above)
Comment (mitza-oci) [reply common/prettify.pm:478]: It is already part of the input data (in the `*.xml` file that defines the contents of the scoreboard)

Comment (iguessthislldo) [reply common/prettify.pm:478]: That works for me.

**Review by iguessthislldo — APPROVED** [review]

**Comment by jwillemsen** [comment]
Get this error when generating our axcioma scoreboard

```
Not enough arguments for Prettify::Process at /home/build/jenkins/workspace/fc_axcioma_scoreboard/autobuild/command/process_logs.pm line 279, near "$log_file)"
Compilation failed in require at /home/build/jenkins/workspace/fc_axcioma_scoreboard/autobuild/autobuild.pl line 83 (#1)
    (F) The function requires more arguments than you specified.
    
Uncaught exception from user code:
	Not enough arguments for Prettify::Process at /home/build/jenkins/workspace/fc_axcioma_scoreboard/autobuild/command/process_logs.pm line 279, near "$log_file)"
	Compilation failed in require at /home/build/jenkins/workspace/fc_axcioma_scoreboard/autobuild/autobuild.pl line 83.
```

**Comment by jwillemsen** [comment]
Command we use is `$WORKSPACE/autobuild/scoreboard.pl -b -n -f $DOC_ROOT/x11-scoreboard/axcioma.xml -y $DOC_ROOT/x11-scoreboard/custom.css -o index.html -d $LOG_ROOT -v`

axcioma.xml can be found at https://github.com/RemedyIT/scoreboard

**Comment by jwillemsen** [comment]
@mitza-oci @kuznetsovmoci Can you have a look at the error I got when using the updated autobuild?

**Comment by kuznetsovmoci** [comment]
I ve created new PR with the fix:

https://github.com/DOCGroup/autobuild/pull/35/files

**Pull request merged**